src/ray/gcs/gcs_client/gcs_client.cc

```diff
@@ -180,15 +180,43 @@ std::pair<std::string, int> GcsClient::GetGcsServerAddress() {
   return current_gcs_server_address_;
 }
 
-void GcsClient::PeriodicallyCheckGcsServerAddress() {
+/// Checks whether GCS at the specified address is healthy.
+bool GcsClient::CheckHealth(const std::string &ip, int port, int64_t timeout_ms) {
+  // Health checking currently needs to be blocking. So it cannot use the event loop
+  // in the GcsClient. Otherwise there can be deadlocks.
+  auto channel = grpc::CreateChannel(absl::StrCat(ip, ":", port),
+                                     grpc::InsecureChannelCredentials());
+  std::unique_ptr<rpc::HeartbeatInfoGcsService::Stub> stub =
+      rpc::HeartbeatInfoGcsService::NewStub(std::move(channel));
+  grpc::ClientContext context;
+  context.set_deadline(std::chrono::system_clock::now() +
+                       std::chrono::milliseconds(timeout_ms));
+  const rpc::CheckAliveRequest request;
+  rpc::CheckAliveReply reply;
+  auto status = stub->CheckAlive(&context, request, &reply);
+  if (!status.ok()) {
+    RAY_LOG(WARNING) << "Unable to reach GCS at " << ip << ":" << port
+                     << ". Failure: " << status.error_code() << " "
+                     << status.error_message();
+    return false;
+  }
+  return true;
+}
+
+void GcsClient::PeriodicallyCheckGcsConnection() {
```

Comment:
Synced offline. I'm good about this change now since we'll refactor (cleanup) this part later during GCS HA. Could you add some todo here? I'll be good to let it go.

Reply:
Added.